gui_agents/s2_5/cli_app.py

```
@@ -15,6 +16,71 @@
+            old_settings = termios.tcgetattr(fd)
+            try:
+                tty.setraw(sys.stdin.fileno())
+                ch = sys.stdin.read(1)
+            finally:
+                termios.tcsetattr(fd, termios.TCSADRAIN, old_settings)
+            return ch
+        else:
+            # Windows fallback
+            import msvcrt
+            return msvcrt.getch().decode('utf-8', errors='ignore')
+    except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)